src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4115,23 +4115,30 @@ private ParameterSyntax ParseParameter()
 
                 TypeSyntax type;
                 SyntaxToken name;
-                SyntaxToken exclamation = null;
+                SyntaxToken exclamationExclamation = null;
                 SyntaxToken equals = null;
                 if (this.CurrentToken.Kind != SyntaxKind.ArgListKeyword)
                 {
                     type = this.ParseType(mode: ParseTypeMode.Parameter);
                     name = this.ParseIdentifierToken();
                     if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
                     {
-                        exclamation = this.EatToken(SyntaxKind.ExclamationToken);
-                    }
-
-                    else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
-                    {
-                        var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
-                        equals = ConvertToMissingWithTrailingTrivia(notEq, SyntaxKind.EqualsToken);
-                        equals = AddError(equals, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
-                        exclamation = SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken);
+                        exclamationExclamation = this.EatToken();
+                        if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                        {
+                            exclamationExclamation = MergeTokens(exclamationExclamation, this.EatToken(), SyntaxKind.ExclamationExclamationToken);
+                        }
+                        else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
```

Comment:
I think you can simplify this set of `if`s: First, check to see if it's an `ExclamationEqualsToken`, and perform the below transformation. Otherwise, unconditionally call `exclamationExclamation = MergeTokens(exclamationExclamation, this.EatToken(SyntaxKind.ExclamationToken, SyntaxKind.ExclamationExclamationToken);`. That should take care of all the relevant cases.